utils/imodels-client-test-utils/src/test-context-providers/imodel/TestIModelRetriever.ts

```diff
@@ -0,0 +1,110 @@
+
+    return {
+      id: iModel.id,
+      name: iModel.name,
+      description: iModel.description!,
+      briefcase,
+      namedVersions,
+      lock
+    };
+  }
+
+  private async queryAndValidateBriefcase(iModelId: string): Promise<BriefcaseMetadata> {
+    const getBriefcaseListParams: GetBriefcaseListParams = {
+      authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
+      iModelId
+    };
+    const briefcases = await toArray(this._iModelsClient.briefcases.getRepresentationList(getBriefcaseListParams));
+    if (briefcases.length !== 1)
+      throw new TestSetupError(`${briefcases.length} is an unexpected briefcase count for reusable test IModel.`);
+
+    return { id: briefcases[0].briefcaseId, deviceName: briefcases[0].deviceName! };
+  }
+
+  private async queryAndValidateNamedVersions(iModelId: string): Promise<NamedVersionMetadata[]> {
+    const getNamedVersionListParams: GetNamedVersionListParams = {
+      authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
+      iModelId
+    };
+    const namedVersions: NamedVersion[] = await toArray(this._iModelsClient.namedVersions.getRepresentationList(getNamedVersionListParams));
+    if (namedVersions.length !== TestIModelCreator.namedVersionIndexes.length)
+      throw new TestSetupError(`${namedVersions.length} is an unexpected named version count for reusable test iModel.`);
+
+    const mappedNamedVersions = namedVersions
+      .map((namedVersion) => ({
+        id: namedVersion.id,
+        changesetId: namedVersion.changesetId!,
+        changesetIndex: this._testIModelFileProvider.changesets.find((cs) => cs.id === namedVersion.changesetId)!.index
+      }))
+      .sort((nv1, nv2) => nv1.changesetIndex - nv2.changesetIndex);
+
+    if (!mappedNamedVersions.every((mappedNamedVersion, i) => mappedNamedVersion.changesetIndex === TestIModelCreator.namedVersionIndexes[i]))
+      throw new TestSetupError("Reusable test iModel contains unexpected named versions.");
+
+    return mappedNamedVersions;
+  }
+
+  private async queryAndValidateLock(iModelId: string): Promise<Lock> {
+    const getLockListParams: GetLockListParams = {
+      authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
+      iModelId
+    };
+    const locks: Lock[] = await toArray(this._iModelsClient.locks.getList(getLockListParams));
+    if (locks.length !== 1)
+      throw new TestSetupError(`${locks.length} is an unexpected lock count for reusable test iModel.`);
+
+    return locks[0];
+  }
+
+  private async findIModelByName(iModelName: string): Promise<IModel | undefined> {
+    const projectId = await this._testProjectProvider.getOrCreate(); // TODO: rename to getOrCreate
+    const iModels = this._iModelsClient.iModels.getRepresentationList({
+      authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
+      urlParams: {
+        projectId
+      }
+    });
+
+    for await (const iModel of iModels) {
```

Comment:
Filtering not implemented in apim?

Reply:
It is implemented now, wasn't before, fixed: https://github.com/iTwin/imodels-clients/pull/65/commits/78ac0f729b248037f56b7736914068f8c3aa44f0